src/boot/efi/boot.c

```diff
@@ -2239,6 +2224,36 @@ static void config_load_xbootldr(
         config_load_entries(config, new_device, root_dir, NULL);
 }
 
+static EFI_STATUS initrd_prepare(const ConfigEntry *entry, CHAR16 **ret_options) {
+        assert(entry);
+        assert(ret_options);
+
+        if (entry->type != LOADER_LINUX || !entry->initrd)
```

Comment:
why the entry-type conditionalization? i.e. the right hand conditionalization should conver that too? i.e. if the type is not LOADER_LINUX then initrd is gonna be NULL anyway, no?

Reply:
You can create an LOADER_EFI config entry with initrds just fine. Does it make any particular sense? Not really. Also, this is just defensive programming.

Reply:
good point